src/util.js

```diff
@@ -216,6 +219,8 @@ function getNextNonSpaceNonCommentCharacter(text, node) {
   return text.charAt(idx);
 }
 
+// This function doesn't ignore trailing comments unlike
+// getNextNonSpaceNonCommentCharacter.
 function getPreviousNonSpaceNonCommentCharacter(text, node) {
```

Comment:
I tried to use `getNextNonSpaceNonCommentCharacter` but couldn't find a way to make it work perfectly to solve this issue.

Kind like `isPreviousLineEmpty`, this doesn't skip trailing comments but should cover the use cases for this issue because any previous trailing comment will just turn the whole node on that line to be a trailing comment. Please let me know if that is ok.